discovery-provider/src/utils/redis_cache.py

```diff
@@ -53,6 +54,24 @@ def use_redis_cache(key, ttl_sec, work_func):
     pickle_and_set(redis, key, to_cache, ttl_sec)
     return to_cache
 
+def get_json_cached_key(redis, key):
```

Comment:
We should deprecate the other one. Less dangerous than this, but can we add a comment to the picke_and_set and get_picked_key that suggests using this json one instead